k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/GenericK8sResourceHandler.kt

```diff
@@ -35,26 +35,18 @@ abstract class GenericK8sResourceHandler <S: GenericK8sLocatable, R: K8sManifest
     override val supportedKind: SupportedKind<S>
         get() = TODO("Not yet implemented")
 
-    override suspend fun toResolvedType(resource: Resource<S>): R =
-        with(resource.spec) {
-            this.template!!.metadata[LABELS].let {
-                val labels : MutableMap<String, String> = if (it == null) {
-                    mutableMapOf<String, String>();
-                } else {
-                    it as MutableMap<String, String>
-                }
-
-                MANAGED_DELIVERY_PLUGIN_LABELS.forEach{ label ->
-                    labels[label.first]  = label.second
-                }
-                this.template!!.metadata[LABELS] = labels
-                (this.template as K8sManifest).spec?.let{ spec ->
-                    augmentWithLabels(spec, MANAGED_DELIVERY_PLUGIN_LABELS)
-                }
-            }
+    @Suppress("UNCHECKED_CAST")
+    override suspend fun toResolvedType(resource: Resource<S>): R {
+        resource.spec.template!!.items?.let {
+            val manifests = it.map{manifest -> labelManifest(manifest)}.toMutableList()
+            resource.spec.template!!.items = manifests
+            return resource.spec.template as R
+        }
 
-            return (this.template as R)
+        with(resource.spec) {
+            return labelManifest(this.template!!) as R
```

Comment:
I know I have used `this.template!!`  in the past. but this could throw an exception. shall we catch and log, if that happens to be the case?

Reply:
I think I would rather have it throw an exception here. If the `template` field is missing, we are screwed because most of our operations need that field.